colossalai/shardformer/policies/bert.py

```diff
@@ -148,35 +143,36 @@ def module_policy(self):
                 target_key=BertModel,
             )
 
-        # Handle bert layer
-        self.append_or_create_submodule_replacement(
-            description=[
-                SubModuleReplacementDescription(
-                    suffix="attention.output.LayerNorm",
-                    target_module=self.Norm,
-                    kwargs={"sp_partial_derived": use_sequence_parallel},
-                ),
-                SubModuleReplacementDescription(
-                    suffix="output.LayerNorm",
-                    target_module=self.Norm,
-                    kwargs={"sp_partial_derived": use_sequence_parallel},
-                ),
-            ],
-            policy=policy,
-            target_key=BertLayer,
-        )
-
-        # handle embedding layer
-        self.append_or_create_submodule_replacement(
-            description=[
-                SubModuleReplacementDescription(
-                    suffix="LayerNorm",
-                    target_module=self.Norm,
-                )
-            ],
-            policy=policy,
-            target_key=BertEmbeddings,
-        )
+        # optimization configuration
+        if self.shard_config.enable_fused_normalization:
+            # Handle bert layer
+            self.append_or_create_submodule_replacement(
+                description=[
+                    SubModuleReplacementDescription(
+                        suffix="attention.output.LayerNorm",
+                        target_module=col_nn.FusedLayerNorm,
+                        kwargs={"sp_partial_derived": use_sequence_parallel},
```

Comment:
how to solver the case where sequence parallel (sp) is enabled, but fused_normalization is not.

Reply:
Having changed the `norm_cls` value in model_policy()